src/Nox.Types/Types/Color/Color.cs

```diff
@@ -74,40 +86,62 @@ public static Color From(string color)
     public static Color FromName(string name)
     {
         var color = System.Drawing.Color.FromName(name);
-        return CreateFromColor(color);
+        return From(color);
+    }
+
+    /// <summary>
+    /// Validates the <see cref="Color"/> object.
+    /// </summary>
+    /// <returns>A validation result indicating whether the <see cref="Color"/> object is valid or not.</returns>
+    internal override ValidationResult Validate()
+    {
+        var result = base.Validate();
+
+        if (!_hexColorRegex.IsMatch(Value))
+        {
+            result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox Color type with unsupported value '{Value}'."));
+        }
+
+        return result;
     }
 
     /// <summary>
     /// Returns the bytes representation of the color value
     /// </summary>
     /// <returns>The byte representation</returns>
-    public byte[] ToBytes() => new[] { SystemColor.A, SystemColor.R, SystemColor.G, SystemColor.B };
+    public byte[] ToBytes()
+    {
+        var color = ToSystemColor();
+        return new[] { color.A, color.R, color.G, color.B };
+    }
 
     /// <summary>
     /// Returns the string representation of the color value, formatted in Hexadecimal format string
     /// </summary>
     /// <returns>The string representation</returns>
-    public string ToHexa() => $"#{SystemColor.A:X2}{SystemColor.R:X2}{SystemColor.G:X2}{SystemColor.B:X2}";
+    public string ToHexa()
```

Comment:
I don't think we need this method, as it can be accessed through the Value property.

Reply:
tohex left, tohexa deleted